streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1146,6 +1146,10 @@ private long advanceNowAndComputeLatency() {
     public void shutdown() {
         log.info("Informed to shut down");
         final State oldState = setState(State.PENDING_SHUTDOWN);
+
+        // If the thread is already shutting down, this is a No-op.
+        if (oldState == State.PENDING_SHUTDOWN || oldState == State.DEAD) return;
```

Comment:
I would like to piggy back this change with this commit. Happy to open a new PR too if you think that is required.

Reply:
I think the condition is never true. Method `setState()` will return `null` if the current state is either `PENDING_SHUTDOWN` (https://github.com/divijvaidya/kafka/blob/1780a0c4676f212b1271688248383420e754718f/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java#L224)  or `DEAD` (https://github.com/divijvaidya/kafka/blob/1780a0c4676f212b1271688248383420e754718f/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java#L230).

Reply:
Even if you check for `null`, `shutdown()` is also without that check a no-op when the thread is already shutting down.

Reply:
Thanks for pointing this out. I missed reading the setState code. I have reverted this change.